### ISSUE TITLE
A kdbx vault acts as a key

### ISSUE THREAD

**Issue by serrq** [Kunzisoft/KeePassDX#1437]
The idea, if properly implemented, makes it possible to transport key and vault files in the same usb storage, securely.

The idea is this:

I create a new database (A) protected only by password and inside I place a key file.

To open the real database (B) I place B in the "vault" field and A in the "key" field. They are both vaults, but one contains only the key (A).

When I press Open the vault that acts as the key will pop up a password prompt to get access to its database. Once the password is entered, database B goes looking for the key file to open its vault.

Advantages of this scheme:

I can carry key file and safe in the usb dongle.
If I am crossing a (potentially dangerous) border I have the option) to remember only one password that the vault that contains the real key.

**Comment by strider72** [comment]
You need to remember *two* passwords. One to unlock the key (A), and another for the vault (B).

**Comment by strider72** [comment]
(I'm not the Dev, just some guy)

This seems like "security by obscurity". No real advantage beyond just having the password directly on your vault. You're trying to make it more secure by complicating the process

**Comment by J-Jamet** [comment]
I agree, it's just complicating the opening, there is no real interest compared to using a password and a key file. The AutoOpen function of KeePassXC has not been implemented yet, it requires the simultaneous opening of several databases, which will be done later. https://keepassxc.org/docs/KeePassXC_UserGuide.html#_automatic_database_opening
